### ISSUE TITLE
Self in sub command

### ISSUE THREAD

**Issue by suntong** [mkideal/cli#20]
Hiya, 

Please take a look at [027s-subc.go](https://gist.github.com/suntong/676211d996b51339813a54b425ac90cc)

This is how I run it, 

```
echo '{ "Dir": "DDD", "Suffix": "sss", "Out": "oooo" }' > build.json

$ go run 027s-subc.go build aaa
[build]:
  &{Helper:{Help:false} Host: Port:80}
  &{Self:0xc8200184c0 Dir:./ Suffix:.go,.c,.s Out:}
  [aaa]

```

The `Self` in sub command is supposed to pick up values defined in the `build.json` file, and I surely remember it did it before, but now it seems not doing it.

**Comment by mkideal** [comment]
I'm sorry. I made a mistake when fix `Reader` and `Writer`, See commit 7bfc6df. And `Self` should be the last field of strcutre:
```go
type buildT struct {
	Dir    string  `cli:"dir" usage:"source code root dir" dft:"./"`
	Suffix string  `cli:"suffix" usage:"source file suffix" dft:".go,.c,.s"`
	Out    string  `cli:"o,out" usage:"output filename"`
	Self   *buildT `cli:"c,config" usage:"config file" json:"-" parser:"jsonfile" dft:"build.json"`
}
```
otherwise, the values(read from json file) will be covered by the `dft` values

**Comment by suntong** [comment]
Bingo. Thanks,
